### PR TITLE
Remove HttpServerTestAdvice

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

File: instrumentation/servlet/servlet-3.0/src/test/groovy/TomcatServlet3Test.groovy [hunk 1]
```diff
@@ -125,22 +123,19 @@ abstract class TomcatServlet3Test extends AbstractServlet3Test<Tomcat, Context>
     }
 
     and:
-    assertTraces(count * 2) {
+    assertTraces(count) {
```
Comment (trask) [inline instrumentation/servlet/servlet-3.0/src/test/groovy/TomcatServlet3Test.groovy:126]: 👍

File: testing-common/src/main/groovy/io/opentelemetry/auto/test/base/HttpServerTest.groovy [hunk 2]
```diff
@@ -479,7 +474,8 @@ abstract class HttpServerTest<SERVER> extends AgentTestRunner {
       attributes {
         "${SemanticAttributes.NET_PEER_PORT.key()}" { it == null || it instanceof Long }
         "${SemanticAttributes.NET_PEER_IP.key()}" { it == null || it == "127.0.0.1" } // Optional
-        "${SemanticAttributes.HTTP_CLIENT_IP.key()}" { it == null || it == TEST_CLIENT_IP } // Optional
+        "${SemanticAttributes.HTTP_CLIENT_IP.key()}" { it == null || it == TEST_CLIENT_IP }
+        // Optional
```
Comment (trask) [inline testing-common/src/main/groovy/io/opentelemetry/auto/test/base/HttpServerTest.groovy:478]:
```suggestion
        // Optional
        "${SemanticAttributes.HTTP_CLIENT_IP.key()}" { it == null || it == TEST_CLIENT_IP }
```

File: instrumentation/grizzly-2.0/src/main/java/io/opentelemetry/auto/instrumentation/grizzly/HttpCodecFilterAdvice.java [hunk 3]
```diff
@@ -43,6 +43,8 @@ public static void onExit(
     HttpRequestPacket httpRequest = (HttpRequestPacket) httpHeader;
     Span span = TRACER.startSpan(httpRequest, httpRequest, method);
 
-    TRACER.startScope(span, ctx);
+    // We don't actually want to attach new context to this thread, as actual request will continue
+    // on some other thread. But we do want to attach that new context to FilterChainContext
+    TRACER.startScope(span, ctx).close();
```
Comment (trask) [inline instrumentation/grizzly-2.0/src/main/java/io/opentelemetry/auto/instrumentation/grizzly/HttpCodecFilterAdvice.java:48]: just curious, is this related to this PR, or a fix for #845?

**Comment by trask** [comment]
I don't know what's going on with CircleCI `muzzle` build:

```
Build-agent version 1.0.38062-115501bf (2020-08-05T09:07:37+0000)
Downloading previous test results from https://circleci.com/gh/open-telemetry/opentelemetry-java-instrumentation/15685

Error downloading previous test results: failed to download from: s3://production-test-results-service/processed-test-results/github.com/open-telemetry/opentelemetry-java-instrumentation/15685/1f892ebd-b7bd-4b32-88a6-35494a4ae59d.json: rpc error: code = Unknown desc = 
```

It's failing with same error in #893 too

File: instrumentation/grizzly-2.0/src/main/java/io/opentelemetry/auto/instrumentation/grizzly/HttpCodecFilterAdvice.java (hunk 3 above)
Comment (iNikem) [reply instrumentation/grizzly-2.0/src/main/java/io/opentelemetry/auto/instrumentation/grizzly/HttpCodecFilterAdvice.java:48]: This test fail on me locally, so I fixed it. It probable is related to that...
